### PR TITLE
Replace obsolete java.util.Date references with Java 8 abstractions (…

### PR DESCRIPTION
**Pull request by cmnbroad**
…java.time).

### CODE REVIEW

**Comment by akiezun** [comment]
@lbergelson please review

File: src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java [hunk 1]
```diff
@@ -145,10 +151,11 @@ public Object instanceMain(final String[] argv) {
         } finally {
             // Emit the time even if program throws
             if (!QUIET) {
-                final Date endDate = new Date();
-                final double elapsedMinutes = (endDate.getTime() - startDate.getTime()) / (1000d * 60d);
+                final ZonedDateTime endDateTime = ZonedDateTime.now();
+                final double elapsedMinutes = (Duration.between(startDateTime, endDateTime).toMillis()) / (1000d * 60d);
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java]: Use `.toMinutes()` instead of toMillis()

Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java]: Sorry, nvm.  Continue doing what you're doing, that would drop the fractional minutes.

**Comment by lbergelson** [comment]
@cmnbroad I'm not very familiar with the new time library.  Why use `ZonedDateTime` in some places and `LocalDateTime` in others?  

Also, what happens with times when we go through daylight savings time? I assume that the new `Duration` machinery handles the difference between "Time" and "Length" for us?

**Comment by cmnbroad** [comment]
I've not used this package before either, but I used ZonedDateTime in the CommandLineProgram code as a way to maintain as much fidelity as possible with the output that was generated using the old classes.  Although LocalDateTime knows the local time, it does not model the Timezone; but the Timezone is required by the DateFormatter for the LONG format. So, although there are other ways to do the formatting to get similar results, it would be at the expense of what I think would be increased code complexity. Also, I could have used ZonedDateTime in the IntegrationTestSpec code as well, but it was unnecessary.

The Duration machinery, as I understand it, is expressly for this kind of purpose,  where you want a duration that is independent of calendar concepts like DST.

**Comment by lbergelson** [comment]
@cmnbroad Thanks for the reply.  That makes sense.  

Looks good to me.

**Comment by lbergelson** [comment]
Welcome to hellbender!

**Comment by cmnbroad** [comment]
Thanks!

**Comment by akiezun** [comment]
@cmnbroad good job. Please close the corresponding issue

**Pull request merged**